gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy

```diff
@@ -127,39 +141,122 @@ class BaselineIdea extends AbstractBaselinePlugin {
+                    def fileName = file.getName()
+                    def destFile = new File(destDir, fileName)
+                    destFile << file.text
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            // Fall back to legacy style file for backwards compatibility
+            // This unfortunately requires to close and reopen IntelliJ after import
+            def ideaStyleFile = project.file("${configDir}/idea/intellij-java-palantir-style.xml")
+            project.file(".idea/codeStyleSettings.xml") << ideaStyleFile.text
+        }
+    }
+
     /**
      * Extracts copyright headers from Baseline directory and adds them to Idea project XML node.
      */
     private void addCopyright(node) {
         def copyrightManager = node.component.find { it.'@name' == 'CopyrightManager' }
         def copyrightDir = Paths.get("${configDir}/copyright/")
-        assert Files.exists(copyrightDir), "${copyrightDir} must exist"
-        def copyrightFiles = project.fileTree(copyrightDir.toFile()).include("*")
-        assert copyrightFiles.iterator().hasNext(), "${copyrightDir} must contain one or more copyright file"
+        def copyrightFiles = getCopyrightFiles(copyrightDir)
         copyrightFiles.each { File file ->
             def fileName = copyrightDir.relativize(file.toPath())
             def copyrightNode = copyrightManager.copyright.find {
                 it.option.find { it.@name == "myName" }?.@value == fileName
             }
             if (copyrightNode == null) {
-                    def copyrightText = XmlUtil.escapeControlCharacters(XmlUtil.escapeXml(file.text.trim()))
-                    copyrightManager.append(new XmlParser().parseText("""
-                        <copyright>
-                            <option name="notice" value="${copyrightText}" />
-                            <option name="keyword" value="Copyright" />
-                            <option name="allowReplaceKeyword" value="" />
-                            <option name="myName" value="${fileName}" />
-                            <option name="myLocal" value="true" />
-                        </copyright>
-                        """.stripIndent()
-                    ))
+                addCopyrightFile(copyrightManager, file, fileName.toString())
             }
         }
 
         def lastFileName = copyrightDir.relativize(copyrightFiles.iterator().toList().sort().last().toPath())
         copyrightManager.@default = lastFileName
     }
 
+    private void addCopyrightIntellijImport() {
+        def copyrightDir = Paths.get("${configDir}/copyright/")
+        def copyrightFiles = getCopyrightFiles(copyrightDir)
+
+        Provider<Node> copyrightManagerNode = new Provider<Node>() {
+            @Override
+            Node get() {
+                return new Node(null, "component", ImmutableMap.of("name", "CopyrightManager"))
+            }
+        }
```

Comment:
can just use a closure, and I'd use Supplier not Provider

```
Supplier<Node> copyrightManagerNode = {
  new Node(null, "component", ImmutableMap.of("name", "CopyrightManager"))
}
```

Reply:
or just create it directly, this doesn't need to be lazy does it?

Reply:
Ah, I was wondering how to make this work in Groovy (I'm very rusty with it)